### PR TITLE
feature/seller

### PR DESCRIPTION
**Pull request by tales-lopes-meli**
Sellers CRUD implementation.

-  FindAll(): List every seller in the system. It must return every data of all of the sellers and a status code  200 in case of success.
- FindOne(**id** int): List seller with **id**. It must return status code 200 in case of success. If seller can't be found, then it will return a 404 return.
- Create(cid int, address string, telephone string, company_name string): Create a new seller in the system. In the success case, it must return a 201 status code and the data of the new seller. If there is a seller with the same cid, than a 409 status code is returned. Also, every field is mandatory, so if one of the data is missing, than a 422 status code is returned.
- Update(**id** int, cid int, address string, telephone string, company_name string): Update a seller with the same id in the path param. In case of success, a 200 status code is returned. If there is a seller with the same cid, than a 409 status code is returned.
- Delete(**id** int): Delete a seller. In case of success, a 204 status code is returned. If the seller is not found, than a 404 status code is returned.

It is already using our error handler pattern.

### CODE REVIEW

File: internal/sellers/service.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import (
+	"errors"
+
+	"github.com/GuiTadeu/mercado-fresh-panic/cmd/server/database"
+	"github.com/imdario/mergo"
+)
+
+var (
+	ExistsSellerCodeError = errors.New("error: seller code already exists")
+	SellerNotFoundError   = errors.New("error: seller not found")
```
Comment (GuiTadeu) [inline internal/sellers/service.go:12]:
```suggestion
	ExistsSellerCodeError = errors.New("Seller code already exists")
	SellerNotFoundError   = errors.New("Seller not found")
```

File: internal/sellers/repository.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	return r.db, nil
+}
+
+func (r *repository) FindOne(id uint64) (database.Seller, error) {
+	for _, seller := range r.db {
+		if seller.Id == id {
+			return seller, nil
+		}
+	}
+	return database.Seller{}, fmt.Errorf("error: seller with id %d not found", id)
```
Comment (GuiTadeu) [inline internal/sellers/repository.go:34]:
```suggestion
	return database.Seller{}, fmt.Errorf("Seller with id %d not found", id)
```

File: internal/sellers/repository.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+func (r *repository) Update(seller database.Seller) (database.Seller, error) {
+	for idx, s := range r.db {
+		if s.Id == seller.Id {
+			r.db[idx] = seller
+			return r.db[idx], nil
+		}
+	}
+	return database.Seller{}, fmt.Errorf("error: seller with id %d not found", seller.Id)
```
Comment (GuiTadeu) [inline internal/sellers/repository.go:51]:
```suggestion
	return database.Seller{}, fmt.Errorf("Seller with id %d not found", seller.Id)
```

File: internal/sellers/repository.go [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+func (r *repository) Delete(id uint64) error {
+	for idx, seller := range r.db {
+		if seller.Id == id {
+			r.db = append(r.db[:idx], r.db[idx+1:]...)
+			return nil
+		}
+	}
+	return fmt.Errorf("error: seller with id %d not found", id)
```
Comment (GuiTadeu) [inline internal/sellers/repository.go:61]:
```suggestion
	return fmt.Errorf("Seller with id %d not found", id)
```

File: cmd/server/database/db.go [hunk 5]
```diff
@@ -4,7 +4,7 @@ import "fmt"
 
 type Seller struct {
```
Comment (GuiTadeu) [inline cmd/server/database/db.go:5]: Não é o ideal mas recomendo colocar os atributos JSON aqui para a response não ir com letra maiúscula
(O certo seria termos uma struct para isso mas daria uma trabalhinho a mais)

**Review by GuiTadeu — APPROVED** [review]

**Review by pedrosouza98 — APPROVED** [review]

**Review by wjpassos — APPROVED** [review]

**Review by anagiorgiani — APPROVED** [review]

**Review by GRMCosta — APPROVED** [review]

File: internal/sellers/service.go (hunk 1 above)
Comment (GuiTadeu) [reply internal/sellers/service.go:12]: Interessante, parece ser algo da linguagem mesmo, não sabia.

Tem até um guia de review do Go que achei aqui que aborda esse ponto: 
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Então acho que aqui podemos manter tudo em minúsculo mas sem o "error", o que acham?
Caso concordem, @tales-lopes-meli poderia aproveitar o PR e corrigir os erros de Section e Products por favor? (eles estão em maiúsculo)

**Pull request merged**